File: tests/journal.lib/channel_inject.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// channel stub
class severity_t : public pyre::journal::channel_t<severity_t> {
    // metamethods
public:
    inline explicit severity_t(const name_type & name) :
        pyre::journal::channel_t<severity_t>(name) {}

    // implementation details
    inline void record() {}
    inline void die() {}
};


// compile time: make sure injection works as expected
int main() {
    // make a channel
    severity_t channel("channel");

    // inject something
    channel
        << pyre::journal::at(__HERE__)
        << pyre::journal::note("time", "now")
        << "hello world!" << pyre::journal::newline
        << pyre::journal::endl;

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/channel_inject.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


# make sure the injection interface works as expected
def test():
    """
    Exercise a simple use case
    """
    # get the channel base class
    from journal.Channel import Channel

    # make one
    d1 = Channel(name="test.channel")

    # build a message
    d1.line("hello world!")
    d1.log()

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
